.github/workflows/release.yml

```diff
@@ -0,0 +1,134 @@
+  GH_TOKEN: ${{ secrets.MERGE_UP_TOKEN }}
+  GIT_AUTHOR_NAME: "DBX PHP Release Bot"
+  GIT_AUTHOR_EMAIL: "dbx-php@mongodb.com"
+  default-release-message: |
+    The PHP team is happy to announce that version {0} of the MongoDB PHP library is now available.
+
+    **Release Highlights**
+
+    TODO: one or more paragraphs describing important changes in this release
+
+    A complete list of resolved issues in this release may be found in [JIRA](https://jira.mongodb.org/secure/ReleaseNote.jspa?version={1}&projectId=12483).
+
+    **Documentation**
+
+    Documentation for this library may be found in the [PHP Library Manual](https://mongodb.com/docs/php-library/current/).
+
+    **Installation**
+
+    This library may be installed or upgraded with:
+    
+      composer require mongodb/mongodb:{0}
+      
+    Installation instructions for the `mongodb` extension may be found in the [PHP.net documentation](https://php.net/manual/en/mongodb.installation.php).
+
+jobs:
+  prepare-release:
+    name: "Prepare release"
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: "Create release output"
+        run: echo '🎬 Release process for version ${{ inputs.version }} started by @${{ github.triggering_actor }}' >> $GITHUB_STEP_SUMMARY
+
+      - uses: actions/checkout@v4
+        with:
+          submodules: true
+          token: ${{ env.GH_TOKEN }}
+
+      - name: "Store version numbers in env variables"
+        run: |
+          echo RELEASE_VERSION=${{ inputs.version }} >> $GITHUB_ENV
+          echo RELEASE_BRANCH=v$(echo ${{ inputs.version }} | cut -d '.' -f-2) >> $GITHUB_ENV
+
+      - name: "Ensure release tag does not already exist"
+        run: |
+          if [[ $(git tag -l ${RELEASE_VERSION}) == ${RELEASE_VERSION} ]]; then
+            echo '❌ Release failed: tag for version ${{ inputs.version }} already exists' >> $GITHUB_STEP_SUMMARY
+            exit 1
+          fi
+
+      - name: "Fail if branch names don't match"
+        if: ${{ github.ref_name != env.RELEASE_BRANCH }}
+        run: |
+          echo '❌ Release failed due to branch mismatch: expected ${{ inputs.version }} to be released from ${{ env.RELEASE_BRANCH }}, got ${{ github.ref_name }}' >> $GITHUB_STEP_SUMMARY
+          exit 1
+
+      #
+      # Preliminary checks done - commence the release process
+      #
+
+      - name: "Set git author information"
+        run: |
+          git config user.name "${GIT_AUTHOR_NAME}"
+          git config user.email "${GIT_AUTHOR_EMAIL}"
+
+      # Create a draft release with a changelog
+      # TODO: Consider using the API to generate changelog
+      - name: "Create draft release with generated changelog"
+        run: gh release create ${{ inputs.version }} --target ${{ github.ref_name }} --generate-notes --draft
+
+      - name: "Read changelog from draft release"
+        run: gh release view ${{ inputs.version }} --json body --template '{{ .body }}' >> changelog
+
+      - name: "Prepare tag message"
```

Comment:
Going back to PHPC's 1.19 release, did we reach a conclusion about not wanting to embed the generated changelog in the tag message? I know we ended up pulling it out of the release notes.

Reply:
I haven't had a chance to update this pull request due to Skunkworks. I've removed the automated changelog, so the release workflow now creates the release with our release message and only uses `Release X.Y.Z` as the tag message. I'll apply those changes to PHPC as well, once I've figured out how to best handle the changelog needed for PECL packaging.